### PR TITLE
Correctness fixes for Pictures

### PR DESCRIPTION
**Pull request by fmatthew5876**
~~Depends on: #1949~~

Fix: #1703 
Fix: #1125
Fix: #1943

### CODE REVIEW

**Comment by fmatthew5876** [comment]
This PR is ready now. I will mark it as "ready for review" once the other picture PRs go in and this gets rebased.

**Comment by fmatthew5876** [comment]
I added some of the optimization commits from #1895 

Retested everything. This is ready to go.

File: src/game_map.h [hunk 1]
```diff
@@ -161,6 +161,12 @@ namespace Game_Map {
 	/** Same as AddScreenX, but for the Y-direction. */
 	void AddScreenY(int& screen_y, int& inc);
 
+	/** @return how many sixteenths we scrolled right during this frame */
+	int GetScrolledRight();
+
+	/** @return how many sixteenths we scrolled right during this frame */
```
Comment (carstene1ns) [inline src/game_map.h]: "down"

Comment (fmatthew5876) [reply src/game_map.h]: Fixed

File: src/game_picture.cpp [hunk 2]
```diff
@@ -342,27 +348,28 @@ void Game_Picture::Update() {
...
-					data.spritesheet_frame = 1;
 				}
 			}
 		}
-
-		data.frames = data.frames + 1;
 	}
 
-	UpdateSprite();
+	if (data.name.empty()) {
```
Comment (carstene1ns) [inline src/game_picture.cpp]: This is now superfluous.

Comment (fmatthew5876) [reply src/game_picture.cpp]: Fixed, thanks

**Comment by fmatthew5876** [comment]
This is fixed now in #1953

**Comment by fmatthew5876** [comment]
jenkins test this please

**Comment by fmatthew5876** [comment]
For the last commit, the trick is to reconstruct all the `Game_Picture` objects on loading the game. The reason we need to do this is because the constructor calls `RequestPictureSprite()`. 

To make this way of doing things work, I also changed the destructor to not erase the pictures, otherwise the clear call would have erased all the pictures in the save data.

Since the destructor was removed, we don't need to explicitly write defaulted move operations anymore.

**Comment by carstene1ns** [comment]
jenkins: test this please

**Pull request merged**